GithubJobsEnterpriseProject/Controllers/GithubJobsController.cs

```diff
@@ -0,0 +1,228 @@
+﻿using GithubJobsEnterpriseProject.Models;
+using GithubJobsEnterpriseProject.Services;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace GithubJobsEnterpriseProject.Controllers
+{
+    [Route("api")]
+    [ApiController]
+    public class GithubJobsController : ControllerBase
+    {
+        private readonly JobContext _context;
+        private HashService _hashService = new HashService();
+        private readonly UserContext _userContext;
+        private readonly IJobApiService _apiService;
+             
+        public GithubJobsController(JobContext context,UserContext userContext, IJobApiService apiService)
```

Comment:
Jobcontext should be abstracted away. Controller must not know of EF Core.
Extract persistence logic into a Repository e.g.: `IJobRepository` -> `JobEfCoreRepository : IJobRepository`
```diff
-public GithubJobsController(JobContext context,UserContext userContext, IJobApiService apiService)
+public GithubJobsController(IJobRepository repository,IUserRepository userRepository, IJobApiService apiService)
```

Reply:
Check repository pattern for further guideline.

Reply:
Create an issue about it and implement it in next iteration.